src/client/options.rs

```diff
@@ -185,74 +206,99 @@ impl FromStr for ServerAddress {
 }
 
 impl ServerAddress {
-    /// Parses an address string into a `ServerAddress`.
+    /// Parses an address string into a [`ServerAddress`].
     pub fn parse(address: impl AsRef<str>) -> Result<Self> {
         let address = address.as_ref();
-        // checks if the address is a unix domain socket
+
         #[cfg(unix)]
         {
             if address.ends_with(".sock") {
-                return Ok(ServerAddress::Unix {
-                    path: PathBuf::from(address),
-                });
+                #[cfg(unix)]
```

Comment:
This is already inside the `#[cfg(unix)]` on line 213, so I don't think this is needed.